### ISSUE TITLE
Can't get "import/no-cycle" to work with eslint flat config

### ISSUE THREAD

**Issue by beedub** [import-js/eslint-plugin-import#2887]
I've been trying to get `import/no-cycle` to work with the new eslint flat config, but to no avail. I'm testing against the very basic example provided in the [doc](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-cycle.md#importno-cycle).

Some notes:
* I _have_ been able to get `import/no-cycle` to work using the non-flat/older style config, so I don't think it's an issue with the project setup
* I also tested `import/max-dependencies` with the flat style config and that _does_ work, so I don't think there's anything obviously broken with my flat config.

I did run `eslint` with `--debug` and did notice that the older style config runs through `file-enumerator` while the newer flat config does not and does not run through anything obviously similar either. It _looks_ like the new flat style config is running rules against each file in isolation rather than considering them as a set together, which would explain why the cycle isn't detected. Is this a bug with `eslint` possibly? or maybe I'm missing some config parameter in the new flat config that changes this?

My `eslint.config.js`:

```js
import importPlugin from "eslint-plugin-import";

export default [
  {
    plugins: {
      'import': importPlugin,
    },
    rules: {
      'import/no-cycle': 'error'
    },
    languageOptions: {
      sourceType: 'module',
      ecmaVersion: 2020,
    },
    ignores: ['eslint.config.js']
  },
]
```

My `.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "plugins": ["import"],
  "parserOptions": {
    "ecmaVersion": 2020,
    "sourceType": "module"
  },
  "rules": {
    "import/no-cycle": ["error"]
  }
}
```

**Comment by ljharb** [comment]
Indeed; it's incompatible, and because eslint doesn't provide any flat-config-based replacement for FileEnumerator, I'm not sure how it can be made compatible.

**Comment by soryy708** [comment]
Flat config landed in v 2.30.0.
How do we move forward with this issue?

**Comment by ljharb** [comment]
I believe this is now fixed with official flat config support. Please file a new issue if not.
